src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1534,12 +1535,28 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument
                 client,
                 resource_group_name,
                 name,
-                kubernetes_version,
+                kubernetes_version='',
                 control_plane_only=False,
                 no_wait=False,
+                node_image_only=False,
                 **kwargs):  # pylint: disable=unused-argument
+    from knack.prompting import prompt_y_n
     instance = client.get(resource_group_name, name)
 
+    if kubernetes_version != '' and node_image_only:
+        raise CLIError('Conflicting flags. Upgrading the Kubernetes version will also upgrade node image version. If you only want to upgrade the node version please use the "--node-image-only" option only.')
+
+    if node_image_only:
+        msg = "This node image upgrade operation will run across every node pool in the cluster and might take a while, do you wish to continue?"
+        if not prompt_y_n(msg, default="n"):
+            return None
+        agent_pool_client = cf_agent_pools(cmd.cli_ctx)
+        for agent_pool_profile in instance.agent_pool_profiles:
+            if agent_pool_profile.type.lower() == "availabilityset":
+                raise CLIError('This cluster is not using VirtualMachineScaleSets. Node image upgrde only operation can only be applied on VirtualMachineScaleSets cluster.')
+            _upgrade_single_agent_pool_node_image(agent_pool_client, resource_group_name, name, agent_pool_profile, no_wait)
```

Comment:
So this is different than our k8 version upgrade in that if one agent pool fails they all fail. This is probably actually what we want to be doing.
Maybe a TODO make them all act similarlily. Then we can be more confident of how upgrades come in rather than promoting it in two different ways. Want to do that but not sure it has to be part of your change.

Reply:
we can discuss it offline about it. That means we can stop PUT MC for agent pool change.